### PR TITLE
76 add an option to make automatic scene switch to a random scene

### PR DESCRIPTION
**Pull request by Th0rstenf**
Now allowing random scene switches as well.

### CODE REVIEW

File: SB_SceneSwitcher/Program.cs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 using System;
 using System.Net.Http;
+using System.Security.Cryptography.X509Certificates;
```
Comment (Th0rstenf) [inline SB_SceneSwitcher/Program.cs:3]: Why is this needed?

File: SB_SceneSwitcher/Program.cs [hunk 2]
```diff
@@ -871,6 +870,39 @@ public void CheckGameStageSong()
...
+                    break;
+                }
+
+                newSongSceneIndex = new Random().Next(0, songScenes.Length);
+            } while (newSongSceneIndex == currentSongSceneIndex);
+
+            currentSongSceneIndex = newSongSceneIndex;
+            itsSceneInterActor.SwitchToScene(songScenes[currentSongSceneIndex], switchScenes);
+        }
+
```
Comment (Th0rstenf) [inline SB_SceneSwitcher/Program.cs:905]: It seems a bit overkill to introduce those, but i can live with it.

**Comment by Th0rstenf** [comment]
I'd highly appreciate if you'd write comments instead of doing changes yourself ;)

**Comment by kozaka-tv** [comment]
I added some more loggs, and extracted the cases to a do action method.
+ removed unnecessary GetMode extension
+ readme updated

File: SB_SceneSwitcher/Program.cs (hunk 1 above)
Comment (kozaka-tv) [reply SB_SceneSwitcher/Program.cs:3]: Idk :) I wonder also why it was in! It is from you! :)

Comment (kozaka-tv) [reply SB_SceneSwitcher/Program.cs:3]: I will delete.

Comment (Th0rstenf) [reply SB_SceneSwitcher/Program.cs:3]: Interesting. Wonder if VS or VSCode is too blame. Didn't insert it myself :D

File: SB_SceneSwitcher/Program.cs (hunk 2 above)
Comment (kozaka-tv) [reply SB_SceneSwitcher/Program.cs:904]: Sorry, but what you mean is overkill?

Comment (Th0rstenf) [reply SB_SceneSwitcher/Program.cs:904]: Introducing  doSequentialSwitch and doRandomSwitch. Both are small, very much self explanatory code snippets and the complexity doesn't justify it imho 🤷‍♂️ 
But as I said, OK for me :)

File: SB_SceneSwitcher/Program.cs [hunk 3]
```diff
@@ -856,11 +842,23 @@ public void CheckGameStageSong()
...
             {
-                if (switchScenes && songSceneAutoSwitchMode.IsOn() &&
-                    itsSceneInterActor.GetTimeSinceLastSceneChange() >= sceneSwitchPeriodInSeconds)
+                CPH.LogDebug("currentScene IsSongScene");
+                CPH.LogVerbose($"songSceneAutoSwitchMode={songSceneAutoSwitchMode}");
+                if (switchScenes && ItsTimeToSwitchScene())
                 {
-                    itsSceneInterActor.SwitchToScene(songScenes[currentSongSceneIndex], switchScenes);
-                    if (++currentSongSceneIndex >= songScenes.Length) currentSongSceneIndex = 0;
+                    switch (songSceneAutoSwitchMode)
```
Comment (kozaka-tv) [inline SB_SceneSwitcher/Program.cs:849]: Maybe this could be an if/else. Because in case we have auto off, or in songScenes only one scene, so this must happen only once during the song.

**Review by kozaka-tv — CHANGES_REQUESTED** [review]

File: SB_SceneSwitcher/Program.cs [hunk 4]
```diff
@@ -871,6 +869,39 @@ public void CheckGameStageSong()
...
+
+            itsSceneInterActor.SwitchToScene(songScenes[currentSongSceneIndex], switchScenes);
+        }
+
+        private void DoRandomSceneSwitch()
+        {
+            int newSongSceneIndex;
+            do
+            {
+                if (songScenes.Length == 1)
```
[the remark this answers is not in the document]
Comment (Th0rstenf) [reply SB_SceneSwitcher/Program.cs:892]: Agreed, we never have to do those switches if there is only one song scene, though it only leads to a forever loop with random switching. I can still extract it and use it as clause before.

However In this case I'd consider it bad, as DoRandomSwitch could then potentially lead to a forever loop if used in a different context....

Therefore I will leave it in DoRandomSwitch as well.

File: SB_SceneSwitcher/Program.cs (hunk 3 above)
Comment (Th0rstenf) [reply SB_SceneSwitcher/Program.cs:849]: It could be, but it has no advantages over a switch.
